### PR TITLE
feature swap to non javax clinical reasoning

### PR DESCRIPTION
**Pull request by Capt-Mac**
closes issue: https://github.com/hapifhir/hapi-fhir/issues/5436

updated testing config for new retrieveSettings and terminologySettings
updated measureOperationProvider (r4/Dstu3) with new Parameter

breaking change to evaluate-measure operation

### CODE REVIEW

**Comment by tadgh** [comment]
Just when I thought i was done reviewing Jakarta migration PRs 👷🏻

**Comment by JPercival** [comment]
This PR should remove the need for the custom CQL/CR branches James used previously for Jakarta. All the needed changes should have already been rolled into those projects.

File: hapi-fhir-storage-cr/pom.xml [hunk 1]
```diff
@@ -139,11 +134,30 @@
 			<version>${spring-security-core.version}</version>
 		</dependency>
 
-		<!-- This is needed for the CqlExceptionHandlingInterceptor -->
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
```
Comment (JPercival) [inline hapi-fhir-storage-cr/pom.xml:138]: issue: Is this dependency necessary? Seems out of place.

File: hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/TestCrConfig.java [hunk 2]
```diff
@@ -98,6 +84,7 @@ public JpaStorageSettings storageSettings() {
 		storageSettings.setAllowExternalReferences(true);
 		storageSettings.setEnforceReferentialIntegrityOnWrite(false);
 		storageSettings.setEnforceReferenceTargetTypes(false);
+		storageSettings.setMaximumExpansionSize(1000);
```
[the remark this answers is not in the document]
Comment (Capt-Mac) [reply hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/TestCrConfig.java:87]: correct, will remove

File: hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/r4/CrResourceListenerTests.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ void testCodeCacheInvalidation() throws InterruptedException {
 		myResourceChangeListenerCacheRefresher.refreshExpiredCachesAndNotifyListeners();
 
 		//cached valueSets
-		assertEquals(11, myEvaluationSettings.getValueSetCache().size());
+		assertEquals(7, myEvaluationSettings.getValueSetCache().size());
```
[the remark this answers is not in the document]
Comment (Capt-Mac) [reply hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/r4/CrResourceListenerTests.java:62]: yes, this was due to expansion change

**Review by barhodes — APPROVED** [review]

File: hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/dstu3/TestCrDstu3Config.java [hunk 4]
```diff
@@ -43,11 +45,31 @@ MeasureEvaluationOptions measureEvaluationOptions(EvaluationSettings theEvaluati
...
 	}
+	@Bean
+	public TerminologySettings terminologySettings(){
+		var termSettings = new TerminologySettings();
+		termSettings.setCodeLookupMode(TerminologySettings.CODE_LOOKUP_MODE.USE_CODESYSTEM_URL);
+		termSettings.setValuesetExpansionMode(TerminologySettings.VALUESET_EXPANSION_MODE.PERFORM_NAIVE_EXPANSION);
+		termSettings.setValuesetMembershipMode(TerminologySettings.VALUESET_MEMBERSHIP_MODE.USE_EXPANSION);
+		termSettings.setValuesetPreExpansionMode(TerminologySettings.VALUESET_PRE_EXPANSION_MODE.USE_IF_PRESENT);
+
+		return termSettings;
```
Comment (tadgh) [inline hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/dstu3/TestCrDstu3Config.java:56]: Are these intended to be non-configurable by users?

Comment (tadgh) [reply hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/dstu3/TestCrDstu3Config.java:56]: Ah, I just realized we are in a test class, disregard!

**Review by tadgh — APPROVED** [review]

**Pull request merged**